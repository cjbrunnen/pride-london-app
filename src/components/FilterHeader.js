import React from "react";
import { View, Text, StyleSheet, StatusBar } from "react-native";
import SafeAreaView from "react-native-safe-area-view";
import {
  headerBgColor,
  filterButtonColor,
  filterButtontextColor
} from "../constants/colors";
import text from "../constants/text";

const FilterHeader = () => (
  <SafeAreaView style={styles.container} forceInset={{ top: "always" }}>
    <StatusBar barStyle="light-content" animated />
    <View testID="filter-header" style={styles.content}>
      <View style={styles.filterButton}>
        <Text style={styles.buttonText}>{text.filterButton}</Text>
      </View>
      <View style={styles.mapButton}>
        <Text style={styles.buttonText}>Map</Text>
      </View>
    </View>
  </SafeAreaView>
);

const styles = StyleSheet.create({
  container: {
    backgroundColor: headerBgColor
  },
  content: {
    alignItems: "center",
    flexDirection: "row",
    paddingHorizontal: 20,
    paddingTop: 8,
    paddingBottom: 12
  },
  filterButton: {
    flex: 1,
    flexDirection: "row",
    height: 43,
    backgroundColor: filterButtonColor,
    padding: 10,
    borderRadius: 4,
    alignItems: "center"
  },
  mapButton: {
    marginLeft: 12,
    width: 52,
    height: 52,
    backgroundColor: filterButtonColor,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 25
  },
  buttonText: {
    color: filterButtontextColor,
    fontWeight: "bold"
  }
});

export default FilterHeader;
