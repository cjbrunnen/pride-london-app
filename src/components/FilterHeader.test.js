// @flow
import React from "react";
import { shallow } from "enzyme";
import FilterHeader from "./FilterHeader";

it("renders correctly", () => {
  const output = shallow(<FilterHeader />);
  expect(output).toMatchSnapshot();
});
