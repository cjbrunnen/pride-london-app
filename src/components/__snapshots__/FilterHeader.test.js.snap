// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<withOrientation
  forceInset={
    Object {
      "top": "always",
    }
  }
  style={
    Object {
      "backgroundColor": "#3A3A3A",
    }
  }
>
  <StatusBar
    animated={true}
    barStyle="light-content"
    showHideTransition="fade"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingBottom": 12,
        "paddingHorizontal": 20,
        "paddingTop": 8,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5C5C5C",
          "borderRadius": 4,
          "flex": 1,
          "flexDirection": "row",
          "height": 43,
          "padding": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ADADAD",
            "fontWeight": "bold",
          }
        }
      >
        I'm interested in...
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5C5C5C",
          "borderRadius": 25,
          "height": 52,
          "justifyContent": "center",
          "marginLeft": 12,
          "width": 52,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ADADAD",
            "fontWeight": "bold",
          }
        }
      >
        Map
      </Text>
    </View>
  </View>
</withOrientation>
`;
